In [2]:
# Libraries
import pandas as pd
from scipy import stats
import matplotlib.pyplot as plt

In [3]:
games = pd.read_csv("../data/processed/game_lifecycle_classified.csv")

## Do growing games have higher review scores than declining games?

H0: There is no difference in average review score between declining and growing games.

H1: There is a difference in average review score between declining and growing games.

In [6]:
declining = games.loc[games["lifecycle_category"] == "Declining","review_score"].dropna()

growing = games.loc[games["lifecycle_category"] == "Growing","review_score"].dropna()

print("Declining games")
print(declining.describe())

print("\nGrowing games")
print(growing.describe())

Declining games
count    1427.000000
mean        0.801951
std         0.134233
min         0.000000
25%         0.740000
50%         0.840000
75%         0.900000
max         0.990000
Name: review_score, dtype: float64

Growing games
count    726.000000
mean       0.841230
std        0.114597
min        0.270000
25%        0.790000
50%        0.870000
75%        0.920000
max        0.980000
Name: review_score, dtype: float64


In [ ]:
# T-test
t_stat, p_value = stats.ttest_ind(
    declining,
    growing,
    equal_var=False
)

print("t-statistic:", t_stat)
print("p-value:", p_value)

t-statistic: -7.087346721256929
p-value: 2.006616177523966e-12


2.01 x 10^-12  < 0.05

Therefore, reject null hypothesis so average review scores differ.

Therefore, growing games were associated with significantly higher review scores than declining games.

## Is multiplayer status associated with a game's 24-month lifecycle outcome?

H0​: Multiplayer status and lifecycle category are independent

H1: Multiplayer status and lifecycle category are associated

In [8]:
games["is_multiplayer"] = (games["tags"].str.contains("Multiplayer", case=False, na=False))

In [9]:
table = pd.crosstab(games["is_multiplayer"], games["lifecycle_category"])

table

lifecycle_category,Declining,Growing,Sustained
is_multiplayer,,,
False,915,329,349
True,512,397,296


In [10]:
from scipy.stats import chi2_contingency

chi2, p_value, dof, expected = chi2_contingency(table)

print("Chi-square statistic:", chi2)
print("P-value:", p_value)
print("Degrees of freedom:", dof)

Chi-square statistic: 72.11833116193753
P-value: 2.1862678389661177e-16
Degrees of freedom: 2


In [11]:
pd.crosstab(
    games["is_multiplayer"],
    games["lifecycle_category"],
    normalize="index"
) * 100

lifecycle_category,Declining,Growing,Sustained
is_multiplayer,,,
False,57.438795,20.652856,21.908349
True,42.489627,32.946058,24.564315


2.19 x 10^-16 < 0.05

Therefore, reject null hypothesis.

Therefore, lifecycle outcome was significantly associated with multiplayer status